features/FEATURE_BLE/ble/UUID.h

```diff
@@ -44,64 +52,103 @@ static uint8_t char2int(char c) {
 public:
+
     /**
-     * Enumeration of the possible types of UUIDs in BLE with regards to length.
+     * Enumeration of the types of UUIDs.
      */
     enum UUID_Type_t {
-        UUID_TYPE_SHORT = 0,    /**< Short 16-bit UUID. */
-        UUID_TYPE_LONG  = 1     /**< Full 128-bit UUID. */
+        /**
+         * 16-bit wide UUID representation.
+         */
+        UUID_TYPE_SHORT = 0,
+
+        /**
+         * 128-bit wide UUID representation.
+         */
+        UUID_TYPE_LONG  = 1
     };
 
     /**
-     * Enumeration to specify byte ordering of the long version of the UUID.
+     * Enumeration of byte ordering.
+     *
+     * It is used to construct 128 byte UUIDs.
      */
     typedef enum {
-        MSB, /**< Most-significant byte first (at the smallest address) */
-        LSB  /**< least-significant byte first (at the smallest address) */
+        /**
+         * Most-significant byte first (at the smallest address).
+         */
+        MSB,
+
+        /**
+         * Least-significant byte first (at the smallest address).
+         */
+        LSB
     } ByteOrder_t;
 
     /**
      * Type for a 16-bit UUID.
      */
-    typedef uint16_t      ShortUUIDBytes_t;
+    typedef uint16_t ShortUUIDBytes_t;
 
     /**
-     * Length of a long UUID in bytes.
+     * Length in bytes of a long UUID.
      */
     static const unsigned LENGTH_OF_LONG_UUID = 16;
+
     /**
      * Type for a 128-bit UUID.
      */
-    typedef uint8_t       LongUUIDBytes_t[LENGTH_OF_LONG_UUID];
+    typedef uint8_t LongUUIDBytes_t[LENGTH_OF_LONG_UUID];
 
     /**
-     * Maximum length of a string representation of a UUID not including the
-     * null termination ('\0'): two characters per
-     * byte plus four '-' characters.
+     * Maximum length for the string representation of a UUID excluding the null
+     * terminator.
+     *
+     * The string is composed of two characters per byte plus four '-'
+     * characters.
      */
     static const unsigned MAX_UUID_STRING_LENGTH = LENGTH_OF_LONG_UUID * 2 + 4;
 
 public:
 
     /**
-     * Creates a new 128-bit UUID.
+     * construct a 128-bit UUID from a string.
      *
-     * @note   The UUID is a unique 128-bit (16 byte) ID used to identify
-     *         different service or characteristics on the BLE device.
+     * @param[in] stringUUID Human readable representation of the UUID following
+     * the format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX.
      *
-     * @param  stringUUID
-     *          The 128-bit (16-byte) UUID as a human readable const-string.
-     *          Format: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
-     *          Upper and lower case supported. Hyphens are optional, but only
-     *          upto four of them. The UUID is stored internally as a 16 byte
-     *          array, LSB (little endian), which is opposite from the string.
+     * @note Upper and lower case are supported.
+     * @note Hyphens are optional and the string must at most includes up to four
```

Comment:
"the string must at most includes up to four of them" -> "the string must include at most four." ?